./script/bootstrap fails on Ubuntu 18.04 (64 bit)

I am trying to build firmware for nrf52840 on Ubuntu 18.04 machine. I executed `./script/bootstrap` as a first step and it failed with following error:

```
  Using cached https://files.pythonhosted.org/packages/b2/34/bfcb43cc0ba81f527bc4f40ef41ba2ff4080e047acb0586b56b3d017ace4/zipp-3.1.0-py3-none-any.whl
Installing collected packages: zipp, importlib-metadata, markdown, tabulate, pygments, mdv
Successfully installed importlib-metadata-1.6.1 markdown-3.2.2 mdv-1.7.4 pygments-2.6.1 tabulate-0.8.7 zipp-3.1.0
./script/bootstrap: 69: ./script/bootstrap: brew: not found
Failed to install shfmt. Install it manually if you need.
darshan@Ubuntu18:~/OT/openthread$
```

It's searching for bew on Ubuntu! Seems like script is for Mac OS. What can be done to fix this?

@jwhui That means ./script/bootstrap is successfull and I can move forward with the build. I tried to build nrf52840 firmware and it fails with following errors (only partial log pasted). Seems like something broken in latest master branch.

```
/home/darshan/OT/openthread/examples/../src/core/common/debug.hpp:79:15: note: in definition of macro 'OT_ASSERT'
         if (!(cond))    \
               ^~~~
In file included from /home/darshan/OT/openthread/examples/../src/core/thread/thread_netif.hpp:40:0,
                 from /home/darshan/OT/openthread/examples/../src/core/common/instance.hpp:73,
                 from /home/darshan/OT/openthread/examples/../src/core/api/channel_manager_api.cpp:37:
/home/darshan/OT/openthread/examples/../src/core/coap/coap_secure.hpp:265:71: error: 'nullptr' was not declared in this scope
     otError SendMessage(Message &aMessage, ResponseHandler aHandler = nullptr, void *aContext = nullptr);
                                                                       ^~~~~~~
/home/darshan/OT/openthread/examples/../src/core/coap/coap_secure.hpp:265:97: error: 'nullptr' was not declared in this scope
     otError SendMessage(Message &aMessage, ResponseHandler aHandler = nullptr, void *aContext = nullptr);
                                                                                                 ^~~~~~~
/home/darshan/OT/openthread/examples/../src/core/coap/coap_secure.hpp:286:60: error: 'nullptr' was not declared in this scope
                         ResponseHandler         aHandler = nullptr,
                                                            ^~~~~~~
/home/darshan/OT/openthread/examples/../src/core/coap/coap_secure.hpp:287:60: error: 'nullptr' was not declared in this scope
                         void *                  aContext = nullptr);
                                                            ^~~~~~~
cc1plus: all warnings being treated as errors
In file included from /home/darshan/OT/openthread/examples/../src/core/meshcop/dataset_manager.hpp:44:0,
                 from /home/darshan/OT/openthread/examples/../src/core/thread/thread_netif.hpp:61,
                 from /home/darshan/OT/openthread/examples/../src/core/common/instance.hpp:73,
                 from /home/darshan/OT/openthread/examples/../src/core/api/channel_manager_api.cpp:37:
/home/darshan/OT/openthread/examples/../src/core/meshcop/dataset.hpp:332:81: error: 'nullptr' was not declared in this scope
     otError ApplyConfiguration(Instance &aInstance, bool *aIsMasterKeyUpdated = nullptr) const;
                                                                                 ^~~~~~~
Makefile:4252: recipe for target 'api/libopenthread_mtd_a-border_router_api.o' failed
make[4]: *** [api/libopenthread_mtd_a-border_router_api.o] Error 1
make[4]: *** Waiting for unfinished jobs....
In file included from /home/darshan/OT/openthread/examples/../src/core/thread/thread_netif.hpp:76:0,
                 from /home/darshan/OT/openthread/examples/../src/core/common/instance.hpp:73,
                 from /home/darshan/OT/openthread/examples/../src/core/api/channel_manager_api.cpp:37:
/home/darshan/OT/openthread/examples/../src/core/thread/address_resolver.hpp:309:56: error: 'nullptr' was not declared in this scope
                              CacheEntryList *  aList = nullptr);
                                                        ^~~~~~~
In file included from /home/darshan/OT/openthread/examples/../src/core/thread/thread_netif.hpp:89:0,
                 from /home/darshan/OT/openthread/examples/../src/core/common/instance.hpp:73,
                 from /home/darshan/OT/openthread/examples/../src/core/api/channel_manager_api.cpp:37:
/home/darshan/OT/openthread/examples/../src/core/utils/child_supervision.hpp: In member function 'ot::Child* ot::Utils::ChildSupervisor::GetDestination(const ot::Message&) const':
/home/darshan/OT/openthread/examples/../src/core/utils/child_supervision.hpp:181:61: error: 'nullptr' was not declared in this scope
     Child *  GetDestination(const Message &) const { return nullptr; }
                                                             ^~~~~~~
cc1plus: all warnings being treated as errors
Makefile:4266: recipe for target 'api/libopenthread_mtd_a-channel_manager_api.o' failed
make[4]: *** [api/libopenthread_mtd_a-channel_manager_api.o] Error 1
Makefile:433: recipe for target 'all-recursive' failed
make[3]: *** [all-recursive] Error 1
Makefile:586: recipe for target 'all-recursive' failed
make[2]: *** [all-recursive] Error 1
Makefile:512: recipe for target 'all' failed
make[1]: *** [all] Error 2
examples/Makefile-nrf52840:329: recipe for target 'do-build-cortex-m4' failed
make: *** [do-build-cortex-m4] Error 2
darshan@Ubuntu18:~/OT/openthread$
```

--- a/script/bootstrap
+++ b/script/bootstrap
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 #
 #  Copyright (c) 2017, The OpenThread Authors.
 #  All rights reserved.
@@ -31,13 +31,17 @@
 #      example compilation and programming.
 #
 
-set -e
+set -euxo pipefail
 
 install_packages_apt()
 {
+    echo 'Installing toolchain dependencies...'
+
     # apt-get update and install dependencies
     sudo apt-get update
     sudo apt-get --no-install-recommends install -y automake g++ libtool lsb-release make cmake ninja-build shellcheck
+
+    echo 'Installing GNU Arm Embedded Toolchain...'
 
     PLATFORM=$(lsb_release -is)
     RELEASE=$(lsb_release -rs)
@@ -56,17 +60,19 @@ install_packages_apt()
         sudo apt-get --no-install-recommends install -y gcc-arm-embedded
     fi
 
+    echo 'Installing pretty tools useful for code contributions...'
+
     # add clang-format for pretty
     sudo apt-get --no-install-recommends install -y clang-format-6.0
 
     # add yapf for pretty
-    python3 -m pip install yapf==0.29.0 || echo 'Failed to install python code formatter yapf. Install it manually if you need.'
+    python3 -m pip install yapf==0.29.0 || echo 'WARNING: could not install yapf, which is useful if you plan to contribute python code to the OpenThread project.'
 
     # add mdv for local size report
-    python3 -m pip install mdv || echo 'Failed to install markdown render. Install it manually if you need.'
+    python3 -m pip install mdv || echo 'WARNING: could not install mdv, which is useful if you plan to contribute markdown to the OpenThread project.'
 
     # add shfmt for shell pretty, try brew only because snap does not support home directory not being /home and doesn't work in docker.
-    command -v shfmt || brew install shfmt || echo 'Failed to install shfmt. Install it manually if you need.'
+    command -v shfmt || brew install shfmt || echo 'WARNING: could not install shfmt, which is useful if you plan to contribute shell scripts to the OpenThread project.'
 }
 
 install_packages_opkg()
@@ -81,8 +87,12 @@ install_packages_rpm()
 
 install_packages_brew()
 {
+    echo 'Installing toolchain dependencies...'
+
     # add build tools
     brew install automake libtool cmake ninja shfmt shellcheck
+
+    echo 'Installing GNU Arm Embedded Toolchain...'
 
     # add ARM toolchain
     brew tap ArmMbed/homebrew-formulae
@@ -93,6 +103,8 @@ install_packages_brew()
         echo 'warning: clang/gcc needed for simulation'
         echo 'warning: please install Command Line Tools from https://developer.apple.com/download/more/'
     fi
+
+    echo 'Installing pretty tools useful for code contributions...'
 
     # add clang-format for pretty
     CLANG_FORMAT_VERSION="clang-format version 6.0"
@@ -128,6 +140,7 @@ install_packages()
 main()
 {
     install_packages
+    echo 'bootstrap completed successfully.'
 }
 
 main

